sdcm/cluster_k8s/__init__.py

```diff
@@ -1353,6 +1359,94 @@ def deploy_node_pool(self, pool: CloudK8sNodePool, wait_till_ready=True) -> None
     def upgrade_kubernetes_platform(self):
         raise NotImplementedError("Kubernetes upgrade is not implemented on this backend")
 
+    @property
+    @contextlib.contextmanager
+    def scylla_config_map(self) -> dict:
+        with self.scylla_config_lock:
+            try:
+                config_map = self.k8s_core_v1_api.read_namespaced_config_map(
+                    name=SCYLLA_CONFIG_NAME, namespace=SCYLLA_NAMESPACE).data or {}
+                exists = True
+            except:  # pylint: disable=bare-except
+                config_map = {}
+                exists = False
+            original_config_map = deepcopy(config_map)
+            yield config_map
+            if original_config_map == config_map:
+                LOGGER.debug("%s: scylla config map hasn't been changed", self)
+                return
+            if exists:
+                self.k8s_core_v1_api.patch_namespaced_config_map(
+                    name=SCYLLA_CONFIG_NAME,
+                    namespace=SCYLLA_NAMESPACE,
+                    body=[{"op": "replace", "path": '/data', "value": config_map}]
+                )
+            else:
+                self.k8s_core_v1_api.create_namespaced_config_map(
+                    namespace=SCYLLA_NAMESPACE,
+                    body=V1ConfigMap(
+                        data=config_map,
+                        metadata={'name': SCYLLA_CONFIG_NAME}
+                    )
+                )
+
+    @contextlib.contextmanager
+    def manage_file_in_scylla_config_map(self, filename: str = 'scylla.yaml') -> ContextManager:
```

Comment:
Consider to don't have default value for `filename` parameter.

Reply:
Done